cmd/controller/main.go

```diff
@@ -501,14 +488,12 @@ func startControllerViaLeaderElection(runOptions controllerRunOptions) error {
 
 		machineController, err := machinecontroller.NewMachineController(
 			runOptions.kubeClient,
-			runOptions.machineClient,
-			runOptions.nodeInformer,
-			runOptions.nodeLister,
-			runOptions.machineInformer,
-			runOptions.machineLister,
-			runOptions.secretSystemNsLister,
+			mgr.GetClient(),
+			mgr.GetRecorder("machine-controller"),
 			runOptions.metrics,
 			runOptions.prometheusRegisterer,
+			runOptions.machineInformer,
+			runOptions.nodeInformer,
```

Comment:
Why do we still need those informers?

Reply:
For triggering the reconciliation. Currently, this controller is not a controller-runtime controller, the only thing the PR changes is that the dynamic client is used everywhere (except on the eviction where that is not possible).

I was thinking this makes the reviewing easier, but I can also extend this PR to make it a controller-runtime controller. WDYT?

Reply:
I would prefer keeping the PR size as is :grimacing: